txpipe/twopoint_null_tests.py

```diff
@@ -766,3 +767,272 @@ def write_output(self, source_list, lens_list, meta, results):
+
+            Qxip = sacc.standard_types.galaxy_shear_xi_plus
+
+            source_tracers = set()
+            for b1, b2 in s.get_tracer_combinations(Qxip):
+                source_tracers.add(b1)
+                source_tracers.add(b2)
+
+            nbin_source = max(len(source_tracers), 1)
+
+            corvarr = np.zeros( (int(nbin_source*(nbin_source+1)/2*self.config["Nell"]), self.config["Nsims"])  )
+
+            ret=np.zeros((nbin_source,nbin_source,self.config["Nell"]))
+
+            c = 0
+            for zi in range(0,nbin_source):
+                for zj in range(zi,nbin_source):
+                    ii = self.config["Nell"]*c
+                    ff = self.config["Nell"]*(c+1)
+                    
+                    # Need all 4 block of the covariance
+                    filename = self.get_input("twopoint_data_real_raw")
+                    tmp      = sacc.Sacc.load_fits(filename)
+                    
+                    # From the sacc file only load relevant tracer combinations 
+                    tmp.keep_selection(tracers=('source_%d'%zj, 'source_%d'%zi))
+                    dvec = tmp.mean
+                    xip  = dvec[:int(self.config["Ntheta"])]
+                    xim  = dvec[int(self.config["Ntheta"]):]
+
+                    #random draws based on mean dn covariance
+                    x = np.random.multivariate_normal(mean = tmp.mean, cov =tmp.covariance.covmat , size = self.config["Nsims"])
+                    Rxip = x[:,:int(self.config["Ntheta"])]
+                    Rxim = x[:,int(self.config["Ntheta"]):]
+
+                    # Compute the uncertainties from random draws          
+                    for n in range(self.config["Nsims"]):
+                        for i in range(int(self.config["Nell"])):
+                            res     = geb_dict['%d'%(i+1)]   
+                            Fp      = res['2']
+                            Fm      = res['3']
+                            En[i,n] = np.sum(Fp*Rxip[:,n] + Fm*Rxim[:,n])/2 
+                            Bn[i,n] = np.sum(Fp*Rxip[:,n] - Fm*Rxim[:,n])/2
+                            
+                    corvarr[ii:ff,:] = Bn
+
+                    # Compute actual data vector
+                    for i in range(int(self.config["Nell"])):
+                        res     = geb_dict['%d'%(i+1)]   
+                        Fp      = res['2']
+                        Fm      = res['3']
+                        En0[i]   = np.sum(Fp*xip + Fm*xim)/2 
+                        Bn0[i]   = np.sum(Fp*xip - Fm*xim)/2
+                        ell[i]  = res['4'][np.argmax(res['5'])]
+
+                    ret[zj,zi,:]= Bn0[:]
+                    c+=1
+                    
+            self.ell     = ell
+            self.results = ret
+            self.cov     = np.cov(corvarr)
+        else:
+            sys.exit("The only method implemented right now are 'hybrideb' or 'namaster_pureB' or 'namaster_nopureB' ")
```

Comment:
You haven't imported `sys`, and in any case it's better to raise a ValueError so that if someone is using this in e.g. a notebook or other interactive use it won't kill their work.

Reply:
Fixed.